eng/liveILLink.targets

```diff
@@ -39,6 +39,12 @@
         <SetTargetFramework Condition="'$(MSBuildRuntimeType)' == 'Core'">TargetFramework=$(NetCoreAppToolCurrent)</SetTargetFramework>
         <SetTargetFramework Condition="'$(MSBuildRuntimeType)' != 'Core'">TargetFramework=$(NetFrameworkToolCurrent)</SetTargetFramework>
     </ProjectReference>
+
+    <!-- Need to reference the analyzer project separately, because there's no easy way to get it as a transitive reference of ILLink.Tasks.csproj -->
+    <ProjectReference Include="$(_ILLinkTasksSourceDir)..\ILLink.RoslynAnalyzer\ILLink.RoslynAnalyzer.csproj"
+                      ReferenceOutputAssembly="false"
+                      OutputItemType="Analyzer"
```

Comment:
Please also set PrivateAssets and Private same as the above ProjectReference sets.

Reply:
Those would probably also benefit from `Private="false"`. That hinders msbuild from copying the source generator into the lib/app output which is unnecessary work. Not 100% sure about PrivateAssets though. Maybe that metadata is by default true when `ReferenceOutputAssembly=false`.

Reply:
OK, I just verified that `PrivateAssets` isn't required here. The ILLink.RoslynAnalyzer.csproj dependency doesn't show up as a package dependency with your changes, i.e. for System.Text.Json:

<img width="364" alt="image" src="https://github.com/dotnet/runtime/assets/7412651/996552aa-302d-41f6-8375-024c3d4137e2">